cgroups/src/common.rs

```diff
@@ -227,3 +227,27 @@ impl PathBufExt for PathBuf {
         Ok(PathBuf::from(format!("{}{}", self.display(), p.display())))
     }
 }
+
+pub(crate) trait StringExt {
+    // this should be replaced by split_once, when we switch to Rust 2021
+    fn split_one(&self, delimiter: &str) -> Option<(&str, &str)>;
+}
+
+impl StringExt for str {
+    fn split_one(&self, delimiter: &str) -> Option<(&str, &str)> {
+        let mut splits = self.splitn(2, delimiter);
+        let first = splits.next()?;
+        let second = splits.next()?;
+
+        Some((first, second))
+    }
+}
+
+#[macro_export]
+macro_rules! hashmap {
```

Comment:
What about this one, available since 1.51?
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=4a06f285da06c18cab7164b9e83bff1f